### DISCUSSION TITLE
Workload certificates not getting rotated after 24 hours

### DISCUSSION THREAD

**Discussion by cer** [istio/istio#44951] [Q&A]
I have a basic install of Istio 1.17.2 (default configuration) on k8s 1.25.2 running in a Kind cluster on a mac.
It works well for 24 hours, e.g. [canary deployments with Flagger](https://docs.flagger.app/tutorials/istio-progressive-delivery)
But after 24 hours it stops working:

```
TLS_error:_268435581:SSL_routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED" 0 0 4 - "-" "-" "-" "-" "10.244.0.39:3306" outbound|3306||customer-service-mysql.default.svc.cluster.local - 10.244.0.39:3306 10.244.0.58:59080 
```

`istioctl proxy-config all` shows this:

```
RESOURCE NAME     TYPE           STATUS     VALID CERT     SERIAL NUMBER                               NOT AFTER                NOT BEFORE
default           Cert Chain     ACTIVE     true           312865665690242056263808614832056594920     2023-05-16T16:21:18Z     2023-05-15T16:19:18Z
ROOTCA            CA             ACTIVE     true           245416724669385952310096386992805491890     2033-05-12T15:25:18Z     2023-05-15T15:25:18Z
```
The `default` certificate has expired - it's `NOT AFTER` time is in the past.
Why is it not getting rotated?

Recreating pods fixes the problem.

Thanks

**Comment by howardjohn** [comment]
The pod is _supposed_ to rotate the certificate after 12 hours. Its not clear why it didn't. Logs from the istio-proxy may show more info (possibly something along the lines of "Failed to generate new certificate" or similar?)

**Reply by howardjohn** [reply]
I did double check to make sure if the CA is down when it tries to rotate that it will indefinitely retry, and it seemed to do so in my case at least.

**Reply by howardjohn** [reply]
If its kind, you may also have suspended your laptop or whatever and the time/clock/etc got messed up. I find even k8s core itself doesn't handle that properly
